dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -134,6 +139,16 @@ def extract_powerschool_data(auth)
     auth
   end
 
+  def extract_microsoft_data(auth)
+    auth_info = auth.info.merge(OmniAuth::AuthHash.new(
+      email: auth[:extra][:raw_info][:userPrincipalName],
```

Comment:
Is this guaranteed to always be the email? Seems like a *very* odd name for it

Reply:
i know. i found it in their [ruby tutorial](https://developer.microsoft.com/en-us/graph/docs/concepts/ruby) and [graph explorer](https://developer.microsoft.com/en-us/graph/graph-explorer).

the graph explorer shows a duplicate field called "mail," but since the ruby reference recommended using "userPrincipalName," i went with that